docs/xwf.1
.\" Written by Rasca Gmelch, Berlin 1999 - 2000
.\" Published under the GNU GPL V2
.TH XWF 1 "Jul. 2000" "GNU" "Version 0.9"
.SH NAME
xwf - gtk+ based filemanager for X Window
.SH SYNOPSIS
.BI "xwf [-v] ["<dir> "] [-g "<x-geometry> "]
.SH OPTIONS
.TP
.BI -v
.br
Verbose mode, give some output on startup.
.TP
.I <dir>
Use this directory as the root for the tree on startup.
.TP
.BI -g \ <x-geometry>
Define the geometry of the toplevel window, e.g.
.I -g 300x400
.SH QUICK REFERENCE
Starting
.I xwf
will use the home-directory as the default root
if no directory is given as argument.

Right mouse button: pops up a context menu (which one depends on selected
items).

Left mouse button: select an item, pressing SHIFT will select
up to and pressing CTRL will add/remove the clicked item. Moving
will initalize a 'Drag and Drop' operation.

Middle mouse button: good for DND operations, cause the selection
will not be changed. The application will also ask you what you
want to do: move, copy or link.

DoubleClick: start the clicked application or if the item is
a normal file start the registered program and give the file
as argument.

Alt+DoubleClick: same as above but start in a terminal program.
If clicked item is a directory, change the root of the tree
to that directory. Concerning a directory you can use also
Ctrl+DoubleClick.

Drag and Drop: the default DND operation is to copy the items.
If you press SHIFT you can move instead of copy selected items.
Using the middle mouse button you will be asked if you want to
copy, move or link the file / directory.
If you DND a URL, e.g. from mozilla a download program is
called and you can watch in a small dialog window the transfer.
.SH KEYBOARD SHORT CUTS
.TP
Alt+A
Select all items in the focused directory.
.TP
Alt+Shift+A
Unselect all items.
.TP
Ctrl+A
Open the 'About' dialog. This gives you the version
information.
.TP
Alt+O
Open marked files or open direcotry in a new window.
.TP
Alt+D
Duplicate the marked file(s) and directories.
.TP
Alt+Shift+D
Hide/show hidden files (dotfiles).
.TP
Alt+E
Empty the trash directory.
.TP
Alt+F
Find files in the current directory.
.TP
Alt+G
Go to selected directory or ask the user if no directory entries
are selected.
.TP
Alt+H
Go to home directory.
.TP
Alt+I
Properties / information of selected items.
.TP
Alt+J
Register an application for a file name suffix.
.TP
Alt+K
Create a new file.
.TP
Alt+N
Create a new directory.
.TP
Alt+M
Make a new alias (soft link).
.TP
Alt+R
Rename selected file / directory.
.TP
Alt+T
Open trash directory in a new window.
.TP
Alt+U
Go one directory level up.
.TP
BACKSPACE
Go one directory level up.
.TP
Alt+W
Close current window.
.TP
Alt+Q
Quit the program.
.TP
Ctrl+X
Delete selected items, same as pressing the Delete key.
.TP
Alt+BACKSPACE
Delete selected itmes but without confirmation dialog.
.SH FILES
.TP
$HOME/.xap/xwf.rc
xwf gtk+ rc file, use it to configure fonts and colors.
.TP
$HOME/.xap/xwf.reg
xwf registry
.TP
$HOME/.xap/.trash/
xwf's trash directory
.TP
$HOME/.xap/.icons/
xwf's user icon directory
.SH SEE ALSO
.B xap(1), xpg(1), X(1)
.SH AUTHOR
(c) Rasca, Berlin
.br
http://home.pages.de/~rasca/
.br
.I xwf
is published under the GNU General Public License



NEWS
0.9.3:
	* xap now supports in it's directories desktop entry files as well
	* many minor bug fixes
	* more mini icons



xwf-src/xwf.c
/*
 * xwf.c
 *
 * Copyright (C) 1999 - 2000 Rasca, Berlin
 * EMail: thron@gmx.de
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 */

#include <stdio.h>
#include <string.h>
#include <unistd.h>
#include <stdlib.h>
#include <sys/stat.h>

#include "config.h"

#ifdef HAVE_GETOPT_H
#include <getopt.h>
#else
extern char *optarg;
extern int optind;
#endif
#ifdef HAVE_LIMITS_H
#include <limits.h>
#endif
#include <X11/Xlib.h>	/* XParseGeometry */
#include "config.h"
#ifdef HAVE_LIBGTK
#include <glib.h>
#include "i18n.h"
#include "xwf_gui.h"
#include "xwf_cfg.h"
#endif
#include "uri.h"

#ifndef VERSION
#define VERSION "(not defined)"
#endif
#define TRASH_DIR ".trash"
#define BASE_DIR ".xap"

int
main (int argc, char *argv[])
{
	int c;
	int verbose = 0;
	int flags = IGNORE_HIDDEN;
	char *path = getenv("HOME"), *geometry = NULL;
	char rc[PATH_MAX+NAME_MAX+1];
	char base[PATH_MAX+1];
	char tmp[PATH_MAX+1];
	char trash[PATH_MAX+1];
	char reg[PATH_MAX+1];
	struct stat st;
	wgeo_t geo = { -1, -1, 400, 480 };

	setlocale (LC_ALL, "");
	bindtextdomain ("xap", LOCALEDIR);
	textdomain ("xap");

	sprintf (rc, "%s/%s/%s", path, BASE_DIR, "xwf.rc");

	gui_init (&argc, &argv, rc);
	while ((c = getopt (argc, argv, "vg:i:")) != EOF) {
		switch (c) {
			case 'v':
				verbose++;
				break;
			case 'g':
				geometry = optarg;
				break;
			case 'i':
				if (atoi(optarg))
					flags |= IGNORE_HIDDEN;
				else
					flags &= ~IGNORE_HIDDEN;
				break;
			default:
				break;
		}
	}
	if (argc != optind) {
		path = argv[argc-1];
	}
	if (strcmp (path, ".") == 0) {
		path = getcwd (NULL, PATH_MAX);
	}
	if (verbose) {
		printf ("Xwf, Version %s\n", VERSION);
		printf ("directory: %s\n", path);
	}
	sprintf (base, "%s/%s", getenv("HOME"), BASE_DIR);
	if (stat(base, &st) == -1) {
		if (verbose) {
			printf ("creating directory: %s\n", base);
		}
		mkdir (base, 0700);
	}
	sprintf (reg, "%s/xwf.reg", base);
	sprintf (trash, "%s/%s", base, TRASH_DIR);
	if (stat (trash, &st) == -1) {
		if (verbose) {
			printf ("creating directory: %s\n", trash);
		}
		mkdir (trash, 0700);
	}
	if (strncmp (path, "..", 2) == 0) {
		sprintf (tmp, "%s/", getcwd(NULL, PATH_MAX));
		strcat (tmp, path);
		path = tmp;
	}
	path = strdup (uri_clear_path(path));
	if (geometry) {
		XParseGeometry (geometry, &geo.x, &geo.y, &geo.width, &geo.height);
		if (verbose) {
			printf ("geometry: %dx%d+%d+%d\n", geo.width, geo.height,
					geo.x, geo.y);
		}
	}
	gui_main (path, base, trash, reg, &geo, flags);
	return 0;
}

